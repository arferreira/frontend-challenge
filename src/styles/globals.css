@import url("https://fonts.googleapis.com/css2?family=Cairo:wght@200;300;400;500;600;700;800;900&display=swap");
@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  height: 100vh;
  background-color: rgb(249, 249, 251);
}
