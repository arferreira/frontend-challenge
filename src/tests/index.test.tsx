import { render } from "@testing-library/react";
import { expect, it, jest } from "@jest/globals";
import Home from "../pages/index";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
}));

jest.mock("next-auth/react", () => {
  const originalModule = jest.requireActual("next-auth/react");
  const mockSession = {
    expires: new Date(Date.now() + 2 * 86400).toISOString(),
    user: { username: "admin" },
  };
  return {
    __esModule: true,
    originalModule,
    useSession: jest.fn(() => {
      return { data: mockSession, status: "authenticated" }; // return type is [] in v3 but changed to {} in v4
    }),
  };
});

//mock use toast
jest.mock("~/providers/ToastProvider", () => ({
  useToast: () => {
    return {
      showToast: jest.fn(),
    };
  },
}));

void it("renders index page unchanged", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
