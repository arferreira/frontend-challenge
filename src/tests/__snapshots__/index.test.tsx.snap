// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="flex flex-col items-center gap-4"
  >
    <div
      class="flex w-full flex-col items-center gap-2"
    >
      <div
        class="flex w-full flex-col items-center p-4 pb-0"
      >
        <h2
          class="items- flex gap-2 text-2xl font-bold text-black dark:text-white-softer"
        >
          Welcome!
        </h2>
        <h3
          class="text-black dark:text-white-softer"
        >
          Hello,
           
          <span
            class="text-blue-600 dark:text-discord"
          />
        </h3>
      </div>
      <div
        class="flex w-full items-center justify-center gap-4 px-4 pb-2"
      >
        <a
          href="/home"
        >
          <button
            class="dark:hover:bg-discord-dark w-full rounded-lg bg-blue-600 px-5 py-2.5 text-center text-sm font-medium text-white transition-all hover:bg-blue-700 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-discord dark:focus:ring-blue-800 sm:w-auto"
          >
            Home
          </button>
        </a>
        <a
          href="/dashboard"
        >
          <button
            class="dark:hover:bg-discord-dark w-full rounded-lg bg-blue-600 px-5 py-2.5 text-center text-sm font-medium text-white transition-all hover:bg-blue-700 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-discord dark:focus:ring-blue-800 sm:w-auto"
          >
            Dashboard
          </button>
        </a>
      </div>
      <div
        class="flex w-full flex-col gap-4 rounded-b-lg bg-white-soft p-4 dark:bg-black-softest"
      >
        <button
          class="dark:hover:bg-discord-dark w-full rounded-lg bg-blue-600 px-5 py-2.5 text-center text-sm font-medium text-white transition-all hover:bg-blue-700 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-discord dark:focus:ring-blue-800 sm:w-auto"
        >
          Sign out
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders index page unchanged 1`] = `
<div>
  <div
    class="flex flex-col items-center gap-4"
  >
    <div
      class="flex w-full flex-col items-center gap-2"
    >
      <div
        class="flex w-full flex-col items-center p-4 pb-0"
      >
        <h2
          class="items- flex gap-2 text-2xl font-bold text-black dark:text-white-softer"
        >
          Welcome!
        </h2>
        <h3
          class="text-black dark:text-white-softer"
        >
          Hello,
           
          <span
            class="text-blue-600 dark:text-discord"
          />
        </h3>
      </div>
      <div
        class="flex w-full items-center justify-center gap-4 px-4 pb-2"
      >
        <a
          href="/home"
        >
          <button
            class="dark:hover:bg-discord-dark w-full rounded-lg bg-blue-600 px-5 py-2.5 text-center text-sm font-medium text-white transition-all hover:bg-blue-700 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-discord dark:focus:ring-blue-800 sm:w-auto"
          >
            Home
          </button>
        </a>
        <a
          href="/dashboard"
        >
          <button
            class="dark:hover:bg-discord-dark w-full rounded-lg bg-blue-600 px-5 py-2.5 text-center text-sm font-medium text-white transition-all hover:bg-blue-700 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-discord dark:focus:ring-blue-800 sm:w-auto"
          >
            Dashboard
          </button>
        </a>
      </div>
      <div
        class="flex w-full flex-col gap-4 rounded-b-lg bg-white-soft p-4 dark:bg-black-softest"
      >
        <button
          class="dark:hover:bg-discord-dark w-full rounded-lg bg-blue-600 px-5 py-2.5 text-center text-sm font-medium text-white transition-all hover:bg-blue-700 focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-discord dark:focus:ring-blue-800 sm:w-auto"
        >
          Sign out
        </button>
      </div>
    </div>
  </div>
</div>
`;
